framework/base/ArrayableTrait.php

```diff
@@ -128,8 +154,54 @@ public function toArray(array $fields = [], array $expand = [], $recursive = tru
     }
 
     /**
+     * Extracts the root field names from nested fields.
+     * Nested fields are separated with dots (.). e.g: "item.id"
+     * The previous example would extract "item".
+     *
+     * @param array $fields The fields requested for extraction
+     * @return array root fields extracted from the given nested fields
+     */
+    public function extractRootFields(array $fields)
+    {
+        $result = [];
+
+        foreach ($fields as $field) {
+            $result[] = current(explode(".", $field, 2));
+        }
+
+        if (in_array('*', $result, true)) {
+            $result = [];
+        }
+
+        return array_unique($result);
+    }
+
+    /**
+     * Extract nested fields from a fields collection for a given root field
+     * Nested fields are separated with dots (.). e.g: "item.id"
+     * The previous example would extract "id".
+     *
+     * @param array $fields The fields requested for extraction
+     * @param string $rootField The root field for which we want to extract the nested fields
+     * @return array nested fields extracted for the given field
```

Comment:
`@since 2.0.14`

Reply:
@cebe @samdark added the "since" in doc and made methods protected